packages/format-currency/src/index.ts

```diff
@@ -28,8 +31,17 @@ export default function formatCurrency(
 	if ( ! currencyDefaults || isNaN( number ) ) {
 		return null;
 	}
-	const { decimal, grouping, precision, symbol } = { ...currencyDefaults, ...options };
+	const { decimal, grouping, precision, symbol, isSmallestUnit } = {
+		...currencyDefaults,
+		...options,
+	};
 	const sign = number < 0 ? '-' : '';
+	if ( isSmallestUnit ) {
+		if ( ! Number.isInteger( number ) ) {
+			return null;
+		}
+		number = convertPriceForSmallestUnit( number, precision );
```

Comment:
What would happen if you override the `precision` value in the `options` object? Would it return an inaccurate result? If that's the case maybe the `currencyDefaults.precision` value should be used here to prevent that scenario.

Reply:
🤔 It would return an accurate result considering the new `precision`, but I see your point. `precision` may have two implied meanings here: the number of decimals used by the currency and the number of decimal places we want to see rendered. 

I _think_ (and I may be wrong) that the intent of the `precision` argument is primarily to override the _currency's_ precision rather than to control how many decimals are displayed because we also have the `stripZeros` option. I suspect that with that option there's no need to actually change `precision` itself. That said, the documentation should clarify that and I'll see if I can do that in this PR.

The only place I think that we're currently overriding `precision` is [here](https://github.com/Automattic/wp-calypso/blob/5508aab1fba60b93fb785b042e422042647d0673/client/my-sites/checkout/composite-checkout/hooks/product-variants.tsx#L39), and I think that `stripZeros` might work better there (but I'll investigate further to be sure).